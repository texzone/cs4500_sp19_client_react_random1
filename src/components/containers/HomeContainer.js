import React from 'react'
import HomeSearchBar from '../views/HomeSearchBar'
import ServiceCategoryPills from '../views/ServiceCategoryPills'
import ServiceProviderService from '../../services/ServiceProviderService'
import { Redirect } from 'react-router-dom';
//import ServiceTabNavigator from './ServiceTabNavigator/ServiceTabNavigator'
//import serviceCategories from '../data/service-categories.mock.json'
class HomeContainer extends React.Component {
  constructor(props) {
      super(props)
      this.serviceProviderService = ServiceProviderService.getInstance()
      this.state = {
        resultProviders: [],
        toProviders: false
      }

    this.searchProviders = this.searchProviders.bind(this)
    this.clearInputs = this.clearInputs.bind(this)

  }

  componentDidMount() {}

  searchProviders() {
    var providerInput = document.getElementById('provider_input').value;
    var zipCodeInput = document.getElementById('zip_code_input').value;
    this.serviceProviderService.
    filterServiceProviders({provider: providerInput, zipCode: zipCodeInput})
        .then(results =>
          this.setState({
            resultProviders: results,
            toProviders: true
          })
        )
  }

  clearInputs() {
    var providerInput = document.getElementById('provider_input');
    var zipCodeInput = document.getElementById('zip_code_input');
    providerInput.value = ""
    zipCodeInput.value = ""
  }

  render() {
      if(this.state.toProviders === true) {
        if(this.state.resultProviders.length > 0) {
          return <Redirect to={{
              pathname: '/services/-1',
              state: { serviceProviders: this.state.resultProviders }
          }}/>
        } else {
          alert('No results found. Please try again.')
          this.clearInputs()
          this.setState({
            toProviders: false
          })
        }
      }
      return (
        <div>
          <div className="row">
              <div className="col-8">
                  <h1>
                      Find professionals near you.
                  </h1>
                  <HomeSearchBar searchFn={this.searchProviders}/>
              </div>
              <div className="col-3 text-right">
                  <a href="#">Sign up</a>
              </div>
              <div className="col-1">
                  <a href="#">Log in</a>
              </div>
          </div>

          <br/>
            <ServiceCategoryPills/>
          <br/>
          <br/>
          <br/>
          <br/>
          <br/>
      </div>
      )

  }
}


export default HomeContainer
