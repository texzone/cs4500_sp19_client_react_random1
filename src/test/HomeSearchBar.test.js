import React from 'react';
import HomeSearchBar from '../components/views/HomeSearchBar';
import HomeContainer from '../components/containers/HomeContainer';
import renderer from 'react-test-renderer';

it('Render HomeSearchBar correctly', () => {
  const tree = renderer.create(
    <HomeSearchBar
    searchFn={HomeContainer.searchProviders}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
