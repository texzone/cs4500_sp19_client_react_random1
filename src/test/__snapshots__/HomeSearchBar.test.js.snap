// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render HomeSearchBar correctly 1`] = `
<div
  className="input-group input-group-lg"
>
  <input
    className="form-control"
    id="provider_input"
    placeholder="Search for providers"
    type="text"
  />
  <input
    className="form-control"
    id="zip_code_input"
    placeholder="Zip code"
    type="text"
  />
  <div
    className="input-group-append"
  >
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      Search
    </button>
  </div>
</div>
`;
